netket/optimizer/jax_stochastic_reconfiguration.py

```diff
@@ -0,0 +1,370 @@
+            elif lsq_solver == "minres":
+                if self._is_holomorphic:
+                    raise RuntimeError(
+                        "minres can be used only for real-valued parameters."
+                    )
+                self._sparse_solver = minres
+            elif not lsq_solver == 'jaxcg':
+                raise RuntimeError("Unknown sparse lsq_solver " + lsq_solver + ".")
+
+        else:
+            if (
+                lsq_solver is None
+                or "ColPivHouseholder" in lsq_solver
+                or "QR" in lsq_solver
+            ):
+                self._lapack_driver = "gelsy"
+            elif "SVD" in lsq_solver or "Cholesky" in lsq_solver:
+                self._lapack_driver = None
+            else:
+                self._lapack_driver = None
+                raise RuntimeError("Unknown lsq_solver" + lsq_solver + ".")
+
+        if self._use_iterative and self._svd_threshold is not None:
+            raise ValueError(
+                "The svd_threshold option is available only for non-sparse solvers."
+            )
+
+    def compute_update(self, oks, grad, out=None):
+        r"""
+        Solves the SR flow equation for the parameter update ẋ.
+
+        The SR update is computed by solving the linear equation
+           Sẋ = f
+        where S is the covariance matrix of the partial derivatives
+        O_i(v_j) = ∂/∂x_i log Ψ(v_j) and f is a generalized force (the loss
+        gradient).
+
+        Args:
+            oks: The matrix of log-derivatives,
+		O_i(v_j)
+            grad: The vector of forces f.
+            out: Output array for the update ẋ.
+        """
+
+        oks -= jax.numpy.mean(oks,axis=0)
+
+        if self.is_holomorphic is None:
+            raise ValueError(
+                "is_holomorphic not set: this SR object is not properly initialized."
+            )
+
+        n_samp = _sum_inplace(_np.atleast_1d(oks.shape[0]))
+
+        n_par = grad.shape[0]
+
+        if out is None:
+            out = _np.zeros(n_par, dtype=_np.complex128)
+
+        if self._is_holomorphic:
+            if self._use_iterative:
+                if self._lsq_solver == "jaxcg":
+                    out = self._jax_cg_solve(oks,grad,n_samp)
+                else:
+                    op = self._linear_operator(oks, n_samp)
+
+                    if self._x0 is None:
+                        self._x0 = _np.zeros(n_par, dtype=_np.complex128)
+
+                    out[:], info = self._sparse_solver(
+                        op,
+                        grad,
+                        x0=self._x0,
+                        tol=self.sparse_tol,
+                        maxiter=self.sparse_maxiter,
+                    )
+                    if info < 0:
+                        raise RuntimeError("SR sparse solver did not converge.")
+
+                self._x0 = out
+            else:
+                self._S = _np.matmul(oks.conj().T, oks, self._S)
```

Comment:
shouldn't we use jax.numpy here (and in all the other cases) instead of standard numpy ?